src/test/java/org/folio/circulation/support/http/client/OffsetTest.java

```diff
@@ -0,0 +1,39 @@
+package org.folio.circulation.support.http.client;
+
+import static org.folio.circulation.support.http.client.Offset.noOffset;
+import static org.folio.circulation.support.http.client.Offset.offset;
+import static org.folio.circulation.support.http.client.PageLimit.limit;
+import static org.folio.circulation.support.http.client.PageLimit.noLimit;
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class OffsetTest {
+  @Test
+  public void shouldCalculateOffsetWhenPageLimitAndOffsetValuesNonNull() {
+    final Offset offset = offset(14).nextPage(limit(7));
+
+    assertEquals(21, offset.getOffset());
+  }
+
+  @Test
+  public void shouldCalculateOffsetWhenOffsetValueIsNull() {
+    final Offset offset = noOffset().nextPage(limit(10));
+
+    assertEquals(10, offset.getOffset());
+  }
+
+  @Test
+  public void shouldCalculateOffsetWhenPageLimitValueIsNull() {
+    final Offset offset = offset(11).nextPage(noLimit());
+
+    assertEquals(11, offset.getOffset());
+  }
```

Comment:
Hmm, wouldn't this result in the same page being fetched repeatedly?

Reply:
Yes, sounds like that.

Reply:
Should this method allow this to happen, to me, it doesn't make sense to ask for the next page with no page size. Should this throw an exception instead?

What do you think?

Reply:
Make sense to me. The method now throws IllegalArgumentException.